packages/immutadot/src/core/curry.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Manage a curried call.
+ * @returns {function|*} Either a new function or the result of the call to the actual function
+ * @this Object
+ */
+function curried(...args) {
+  const { fn, last, minArity } = this
+
+  if (last) {
+    const obj = args.length === 0 ? undefined : args[0]
```

Comment:
Why do you check for the length of the array ?

```es6
const [ obj ] = args
```

This doesn't achieve the same result ?

Reply:
Yes, somehow I forgot how "flexible" JavaScript is...